base/sparse/sparsematrix.jl

```diff
@@ -17,12 +17,16 @@ immutable SparseMatrixCSC{Tv,Ti<:Integer} <: AbstractSparseMatrix{Tv,Ti}
         n < 0 && throw(ArgumentError("number of columns (n) must be ≥ 0, got $n"))
         new(Int(m), Int(n), colptr, rowval, nzval)
     end
+    SparseMatrixCSC(m::Integer, n::Integer) = SparseMatrixCSC{Tv,Ti}(m, n, ones(Ti, max(n,0)+1), Array{Ti}(0), Array{Tv}(0))
```

Comment:
That duplicates the check from the constructor above which it delegates to, but is probably better than the current code, I agree. Maybe even better to have a dedicated function to check all sizes are non-negative, as this check should occur in all the array constructors at some point.